doc/acl/Dynamic-ACL-Design.md

```diff
@@ -0,0 +1,190 @@
+                        "acl-entry":{
+                            "1":{
+                                "actions":{
+                                    "config":{
+                                        "forwarding-action":"ACCEPT"
+                                    }
+                                },
+                                "config":{
+                                    "sequence-id":1
+                                },
+                                "ip":{
+                                    "config":{
+                                        "source-ip-address":"192.168.0.2/32",
+                                        "destination-ip-address":"192.168.0.3/32"
+                                    }
+                                },
+                                "ttl":"300"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+The YANG of ACL_RULE is required to be updated to accept new field `is_dynamic`
+
+Orchagent (actually `aclorch`) won't consume the value of `is_dynamic`. So no change is required to `orchagent`.
+#### STATE DB
+A new table is introduced to state_db to record the timestamp of creation, expiration and TTL of an ACL rule.
+```
+  key                      = ACL_TTL_TABLE:acl_rule_name     ; acl_rule_name specifies the corresponding ACL rule name, must be unique
+  ;field                   = value  
+  creation_time            = Integer                         ; timestamp when the rule is created or refreshed  
+  expiration_time          = Integer                         ; timestamp when the rule will expire
+  ttl                      = Integer                         ; the ttl, in second 
+```
+A sample
+```json
+{
+    "ACL_TTL_TABLE|DATAACL|RULE_1":{
+        "creation_time":1645686213,
+        "expiration_time":1645686513,
+        "ttl":300
+    }
+}
+```
+#### acl-loader
+
+Update `acl-loader` script to parse new field `is_dynamic`. The entry will be created in `ACL_TTL_TABLE` in `state_db` if `ttl` is present for certain an ACL rule in `acl.json`. Please find more details in workflow diagram.  
+
+#### acl_ttl_checker
+
+A helper script will be added to `swss` container. The checker is started after `orchagent` and check the TTL of dynamic ACL rules every 10 seconds by default. It will walk through all entries in `ACL_TTL_TABLE` and delete the corresponding `ACL_RULE` from `config_db` if
+
+- Current timestamp is larger than the `expiration_time` in the entry
+- The corresponding ACL rule is marked as dynamic
+
+### Work flow
+#### Add dynamic ACL rule
+<p align=center>
+<img src="img/dynamic_acl_creation.png" alt="Figure 1. Create dynamic ACL rule workflow">
+</p>
+
+#### Refresh TTL of existing ACL rule
+<p align=center>
+<img src="img/dynamic_acl_refresh.png" alt="Figure 2. Refresh dynamic ACL rule workflow">
+</p>
+
+#### Remove ACL rule when TTL expires
+<p align=center>
```

Comment:
What if user does a 'config save" during the time acl rule is active? A future config reload can re-install the rule.